src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -63,7 +63,8 @@ public static ConsoleColor BackgroundColor
 
         public static void ResetColor()
         {
-            if (!ConsoleOutIsTerminal)
+            // We only want to reset colors if we're targeting a TTY device
+            if (Console.IsOutputRedirected)
```

Comment:
We actually want the IsXxRedirected behavior to be the same across the platforms, right?  In which case SetOut should not affect it on Unix either?

Seems like we have two cases:
- IsXxRedirected checks the file descriptors for 0/1/2 to see if they're character devices
- Color ANSI color checking checks the file descriptors underlying the output stream to see if they're character devices